### DISCUSSION TITLE
New procedures in module interface - adding extended interface

### DISCUSSION THREAD

**Discussion by makarukp** [thesofproject/sof#6926] [Ideas]
### Current state
We are migrating from component API to module API. Current module interface procedures set is very reduced and hard to match specific module requirements.
There are some native modules which require specific actions on module state transition.  In such case the only reasonable way now is to use generic set_configuration procedure to make call for any specific module function. It seems to be the only way for simple communication, but it is complex and not meaningful at the same time. 

For Example some modules may require calling simple functions on bind/unbind or changing pipeline states. In module interface there is no clear way to express such action. 

### Idea
To implement those additional procedures in extended module interface  to emphesize any module action which has to be taken on module state transition. Extended module interface could contain basic module interface structure compatible with IADK, so they could be interchangably used depending on type of module.

### Motivation
Module interface clarification for native fw modules.

**Comment by plbossart** [comment]
@makarukp I would suggest rewriting your suggestion taking into account that the word 'trigger' is an overloaded term. For Linux/ALSA, 'trigger' means starting/stopping/pausing/releasing the data stream and not perform ANY configuration. Trigger is supposed to be a fast operation with limited IPC and preferably none.

**Reply by lgirdwood** [reply]
@makarukp any high level proposal ready in the form of a patch ?
@ranj063 any inputs ?
